docs/Guides/Migration-Guide-V4.md

```diff
@@ -16,6 +16,17 @@ have decided not to support the use of middlewares. Both
 [`@fastify/express`](https://github.com/fastify/fastify-express) will still be
 there and maintained. Use Fastify's [hooks](../Reference/Hooks.md) instead.
 
+### `reply.res` moved to `reply.raw`
+
+If you previously used the `reply.res` attribute to access the underlying Request
+object you'll instead need to depend on `reply.raw`.
+
+### Need to `return reply` in some instances
```

Comment:
```suggestion
### Need to `return reply` to signal a "fork" of the promise chain
```

We should be a bit more explicit and express _why_ we have done this. There was a lot of ambiguity and a few bad edge cases that we could not handle before.

The recommended approach is to return the payload or stream instead of using reply.send().